src/Client.php

```diff
@@ -59,19 +59,16 @@ class Client extends Model
     public $plainSecret;
 
     /**
-     * Bootstrap the model and its traits.
+     * Create a new Eloquent model instance.
      *
+     * @param  array  $attributes
      * @return void
      */
-    public static function boot()
+    public function __construct(array $attributes = [])
     {
-        parent::boot();
+        parent::__construct($attributes);
 
-        static::creating(function ($model) {
-            if (Passport::clientUuids()) {
-                $model->{$model->getKeyName()} = $model->{$model->getKeyName()} ?: (string) Str::orderedUuid();
-            }
-        });
+        $this->usesUniqueIds = Passport::clientUuids();
     }
```

Comment:
We don't recommend overwriting a model's constructor. Can we do this somehow else? maybe with a `bootingX` method or something?

Reply:
boot methods are static but `$this->usesUniqueIds` is not, we can use `initializeX` method, are you ok with defining a new trait for this? tbh, I personally prefer to get rid of `Passport::clientUuids()` totally.

Reply:
We can't really get rid of that method unless we require older apps to fill out that database column which is quite a bit of a hassle for them. I guess we can give this one a shot.

Reply:
@driesvints actually there is no need to fill out that DB column, `string(36)` can be used if they were using incremental ids, in this case they can keep old ids, and get uuid for new clients. what you think?

Reply:
> unless we require older apps to fill out that database column

Not necessarily, they can keep the old IDs as is, and get uuid for new clients by using `string(36)` column type.